### ISSUE TITLE
[Expo] waitForElement: Promise constructor's argument is not a function

### ISSUE THREAD

**Issue by wKovacs64** [callstack/react-native-testing-library#29]
Created a fresh Expo app and tried to use `waitForElement`, but it throws. Works fine in vanilla RN app.

Test:
```js
it('renders', async () => {
  const { getByText } = render(<App />);

  await waitForElement(() => getByText(/working/));
});
```
Result:
```
TypeError: Promise constructor's argument is not a function

     7 |     const { getByText } = render(<App />);
     8 | 
  >  9 |     await waitForElement(() => getByText(/working/));
    10 |   });
    11 | });
    12 | 
    
    at new Promise (node_modules/promise/lib/core.js:59:11)
    at resolve (node_modules/react-native-testing-library/src/waitForElement.js:8:26)
    at Object.getByText (App.test.js:9:32)
    at tryCatch (node_modules/regenerator-runtime/runtime.js:62:40)
    at Generator.invoke [as _invoke] (node_modules/regenerator-runtime/runtime.js:296:22)
    at Generator.prototype.(anonymous function) [as next] (node_modules/regenerator-runtime/runtime.js:114:21)
    at tryCatch (node_modules/regenerator-runtime/runtime.js:62:40)
    at invoke (node_modules/regenerator-runtime/runtime.js:152:20)
    at node_modules/regenerator-runtime/runtime.js:195:11
    at tryCallTwo (node_modules/promise/lib/core.js:45:5)
```

I'm assuming it has something to do with Expo's [jest-expo](https://github.com/expo/expo/tree/master/packages/jest-expo) preset, but I haven't narrowed it down yet.

Reproduction repo if anyone wants to mess with it: https://github.com/wKovacs64/rntl-expo-test

**Comment by thymikee** [comment]
Based on stack trace looks like expo preset sets global promise to a npm promise module. Will have to investigate why it fails, but I have no access to my workstation this weekend

**Comment by wKovacs64** [comment]
Looks like `react-native` itself sets the global promise in its [Jest setup script](https://github.com/facebook/react-native/blob/95174d4ea8d050305885f3cc2a4c99e2b83fc961/jest/setup.js#L21), and `jest-expo` includes that script in its `setupFiles` (as well as its own). But just glancing around `react-native-testing-library`, it doesn't seem like you're doing anything crazy that wouldn't work with the `promise` module (?).

**Comment by wKovacs64** [comment]
Fixed by #31, in case anyone stumbles upon this later.
